src/bp/core/services/dialog/flow/service.ts

```diff
@@ -66,20 +120,31 @@ export class FlowService {
   }
 
   private _listenForCacheInvalidation() {
-    this.cache.events.on('invalidation', (key: string) => {
-      const matches = key.match(/\/bots\/([A-Z0-9-_]+)\/flows\//i)
-      if (matches && matches.length >= 1) {
+    this.cache.events.on('invalidation', async (key: string) => {
+      if (this._flowCache.empty()) {
+        return
+      }
+
+      const matches = key.match(/object::[\s\S]+\/bots\/([A-Z0-9-_]+)\/flows\/([\s\S]+(flow|ui)\.json)/i)
+      if (matches && matches.length >= 2) {
         const botId = matches[1]
-        if (this._allFlows.has(botId)) {
-          this._allFlows.delete(botId)
+        const flowPath = this.toFlowPath(matches[2])
```

Comment:
_allFlows was already meant to be a cache. What improvement does your class brings ? Is the class really necessary ? It only seems to bring a bit more complexity :/

Reply:
The main difference is that the old cache was cleared of all its flows every time a single flow was inserted, updated, deleted or renamed (e.g. [service.ts](https://github.com/botpress/botpress/blob/master/src/bp/core/services/dialog/flow/service.ts#L219)). Meaning that it served as a cache but only as long as nothing changed (making it practically useless imho).

Reply:
Ah, good point ! Please explain the reason of the change in the description, not only the new behavior :)